app/presenters/admin_access_presenter.rb

```diff
@@ -18,31 +18,47 @@ def permitted_access_levels
     UserAccess::LEVELS.slice(*admin.permitted_access_levels).values
   end
 
-  memoize def access_tree
-    UserAccessTree.new(admin)
-  end
-
-  delegate :visible?, to: :access_tree
-
   def visible_access_tree
     if admin.access_across_organizations?(:any)
       {
-        data: access_tree.organizations,
+        data: organization_tree,
         render_partial: "email_authentications/invitations/organization_access_tree",
         root: :organization
       }
     elsif admin.access_across_facility_groups?(:any)
       {
-        data: access_tree.facility_groups,
+        data: facility_group_tree,
         render_partial: "email_authentications/invitations/facility_group_access_tree",
         root: :facility_group
       }
     else
       {
-        data: access_tree.facilities,
+        data: visible_facilities,
         render_partial: "email_authentications/invitations/facility_access_tree",
         root: :facility
       }
     end
   end
+
+  memoize def organization_tree
+    facility_group_tree
+      .group_by { |facility_group, _facilities| facility_group.organization }
+      .transform_values(&:to_h)
+  end
+
+  memoize def facility_group_tree
+    visible_facilities.group_by(&:facility_group)
+  end
+
+  memoize def visible_organizations
+    admin.accessible_organizations(:any)
+  end
+
+  memoize def visible_facility_groups
+    admin.accessible_facility_groups(:any)
+  end
+
+  memoize def visible_facilities
+    admin.accessible_facilities(:any)
```

Comment:
This isn't related to this PR, but could you shed some light on what the `any` action does, and how it should be used?